### ISSUE TITLE
Comments at the end of a source file are ignored

### ISSUE THREAD

**Issue by vitaly-t** [jsdoc3/jsdoc#1396]
External links are no longer generated after upgrading to v3.5

**Comment by hegemonic** [comment]
I'm not sure exactly what this means. Can you provide a code sample that demonstrates this issue and explain what isn't showing up in the generated docs?

**Comment by vitaly-t** [comment]
If we define an external link like this:

```js
/**
 * @external Promise
 * @see https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise
 */
```
It no longer generates an HTML file that represents the link, and when trying to use it:

```js
@property {external:Promise} result
```

It no longer generates a link.

**Comment by hegemonic** [comment]
I'm not able to reproduce this issue. I created a new file with these contents:

```js
/**
 * @external Promise
 * @see https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise
 */
```

When I run `jsdoc path/to/file.js`, it generates the file `./out/external-Promise.html`, just as I would expect.

If you follow those steps, are you able to reproduce this issue?

**Comment by vitaly-t** [comment]
Interesting, but it doesn't work here. I've tried many times, upgrading to v3.5 and back to v3.4, and just as I switch over to v3.5, the external files are no longer generated.

**Comment by vitaly-t** [comment]
The project where I'm testing it is [pg-promise](https://github.com/vitaly-t/pg-promise).

**Comment by hegemonic** [comment]
Okay, I'm able to reproduce the issue when I generate docs for `pg-promise`. Investigating.

**Comment by hegemonic** [comment]
Looks like switching to the Babylon parser caused this issue. I'm guessing that it does something tricky with comment attachment that we need to work around.

**Comment by hegemonic** [comment]
This issue only occurs when the `estree` plugin for Babylon is enabled, and there's a `'use strict';` declaration at the top of the file. In this case, all of the trailing comments at the end of the file are erroneously attached to the `ExpressionStatement` node for the `'use strict';` directive, rather than the last node in the `Program` node's body. [This example](https://astexplorer.net/#/gist/3ab9c8567c9d0a9679e5055eac3d2471/15067337efba599e4417dff5976734d8e2b7eb86) demonstrates the problem.

The only way I can work around this in JSDoc is to stop using the `estree` plugin, which would require a lot of changes to JSDoc. So I'll see if I can get the Babylon folks to fix their bug.

In the meantime, if anyone else is running into this issue, you can work around it by making sure you don't have any JSDoc comments after the last line of code in your JS file.

**Comment by hegemonic** [comment]
Okay, I filed babel/babylon#619. If it doesn't get any traction, I'll look into disabling the `estree` plugin.

**Comment by vitaly-t** [comment]
@hegemonic Thank you for investigating it! 👍 

> In the meantime, if anyone else is running into this issue, you can work around it by making sure you don't have any JSDoc comments after the last line of code in your JS file.

Unfortunately, this is the only logical place where to place them 😭

**Comment by hegemonic** [comment]
On second thought, I can't disable the `estree` plugin, because that would break things for anyone who's using an [AST node visitor plugin](http://usejsdoc.org/about-plugins.html#node-visitors). So we'll have to wait for Babylon to fix the bug.

**Comment by hegemonic** [comment]
I managed to hack my way around the Babylon bug, so this issue is now fixed! The fix will be included in JSDoc 3.5.3.
